exporter/awsemfexporter/metric_translator.go

```diff
@@ -159,10 +168,101 @@ func TranslateOtToCWMetric(rm *pdata.ResourceMetrics, dimensionRollupOption stri
 				continue
 			}
 			cwMetrics := getCWMetrics(&metric, namespace, instrumentationLibName, dimensionRollupOption)
-			cwMetricList = append(cwMetricList, cwMetrics...)
+			cwMetricLists = append(cwMetricLists, cwMetrics...)
+			batchCWMetrics(cwMetrics, groupedCWMetricMap, cwMetricsMap)
 		}
 	}
-	return cwMetricList, totalDroppedMetrics
+	return cwMetricLists, totalDroppedMetrics
+}
+
+func batchCWMetrics(cwMetricList []*CWMetrics, groupedCWMetricMap map[string]*GroupedCWMetric, cwMetricsMap map[string]*CWMetrics) {
+	for _, met := range cwMetricList {
+		if len(met.Measurements[0].Dimensions[0]) > 0 {
+			key := strings.Join(met.Measurements[0].Dimensions[0], "")
```

Comment:
I don't think you should use the dimensions as the key. Once we have the metric declaration logic, dimensions != labels. I think this ties in to my earlier comment of performing batching before dimension setting.